deps.ffmpeg/40-x264.ps1

```diff
@@ -3,17 +3,24 @@ param(
     [string] $Version = 'r3106',
     [string] $Uri = 'https://github.com/mirror/x264.git',
     [string] $Hash = 'eaa68fad9e5d201d42fde51665f2d137ae96baf0',
-    [array] $Targets = @('x64'),
+    [array] $Targets = @('x64', 'arm64'),
     [switch] $ForceShared = $true
 )
 
 function Setup {
     Setup-Dependency -Uri $Uri -Hash $Hash -DestinationPath $Path
 
     if ( ! ( $SkipAll -or $SkipDeps ) ) {
-        Invoke-External pacman.exe -S --noconfirm --needed --noprogressbar nasm
+        if ( $Target -ne 'arm64' ) {
+            Invoke-External pacman.exe -S --noconfirm --needed --noprogressbar nasm
+        }
         Invoke-External pacman.exe -S --noconfirm --needed --noprogressbar make
     }
+
+    if ( $Target -eq 'arm64' ) {
+        Remove-Item -Path "${Path}/tools/gas-preprocessor.pl" -ErrorAction SilentlyContinue
+        Copy-Item -Path "$($script:WorkRoot)/gas-preprocessor/gas-preprocessor.pl" -Destination "${Path}/tools/"
+    }
```

Comment:
Would this be more appropriate in the `Patch()` function?

Reply:
Using a patch for this would remove the ability to automatically update the preprocessor with an updated version from its own git repository. This way we'd just have to update the gas-preprocessor commit hash.

Reply:
Fair.